sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileClient.cs

```diff
@@ -2825,6 +2915,52 @@ Response<Blobs.Models.BlobDownloadInfo> response
                 conditions: overwrite ? null : new DataLakeRequestConditions { IfNoneMatch = new ETag(Constants.Wildcard) },
                 cancellationToken: cancellationToken);
 
+        /// <summary>
+        /// The <see cref="Upload(string, DataLakeFileUploadOptions, CancellationToken)"/>
+        /// operation creates and uploads content to a file.  If the file already exists, its content will be overwriten.
+        ///
+        /// For more information, see <see href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/update" />.
+        /// </summary>
+        /// <param name="path">
+        /// A file path containing the content to upload.
+        /// </param> of this new block blob.
+        /// <param name="options">
+        /// Optional parameters.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{BlobContentInfo}"/> describing the
+        /// state of the updated block blob.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public virtual Response<PathInfo> Upload(
+            string path,
+            DataLakeFileUploadOptions options,
+            CancellationToken cancellationToken = default)
+        {
+            using (FileStream stream = new FileStream(path, FileMode.Open))
+            {
+                return StagedUploadAsync(
+                    stream,
+                    options?.HttpHeaders,
+                    options?.Metadata,
+                    options?.Permissions,
+                    options?.Umask,
+                    options?.Conditions,
+                    options?.ProgressHandler,
+                    options.TransferOptions,
```

Comment:
".?"

Reply:
Old habit, will remove.

Reply:
Actually I think we need the .?.  The user can pass in null for DataLakeFileUploadOptions